exasol_transformers_extension/udfs/sequence_classification_single_text_udf.py

```diff
@@ -0,0 +1,152 @@
+                bucketfs_location=bucketfs_location,
+                model_path=f"container/{model_path}")
+
+    def load_models(self, model_name: str) -> None:
+        """
+        Load model and tokenizer model from the cached location in bucketfs
+
+        :param model_name: The model name to be loaded
+        """
+        self.last_loaded_model = self.base_model.from_pretrained(
+            model_name, cache_dir=self.cache_dir)
+        self.last_loaded_tokenizer = self.tokenizer.from_pretrained(
+            model_name, cache_dir=self.cache_dir)
+        self.last_loaded_model_name = model_name
+
+    def get_prediction(self, model_df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Perform prediction of the given model and preparation of the prediction
+        results according to the format that the UDF can emit.
+
+        :param model_df: The dataframe to be predicted
+
+        :return: The dataframe where the model_df is formatted with the
+        prediction results
+        """
+        preds, labels = self._predict_model(list(model_df['text_data']))
+        pred_df = self._prepare_prediction_dataframe(model_df, preds, labels)
+        return pred_df
+
+    def _predict_model(self, sequences: List[str]) -> \
+            Tuple[List[float], List[str]]:
+        """
+        Predict the given text list using recently loaded models, return
+        probability scores and labels
+
+        :param sequences: The list of text to be predicted
+
+        :return: A tuple containing prediction score list and label list
+        """
+        tokens = self.last_loaded_tokenizer(sequences, return_tensors="pt")
+        logits = self.last_loaded_model(**tokens).logits
+        preds = torch.softmax(logits, dim=1).tolist()
+        labels_dict = self.last_loaded_model.config.id2label
+        labels = list(map(lambda x: x[1], sorted(labels_dict.items())))
+
+        return preds, labels
+
+    @staticmethod
+    def _prepare_prediction_dataframe(
+            model_df: pd.DataFrame, preds: List[float], labels: List[str]) \
+            -> pd.DataFrame:
+        """
+        Reformat the dataframe used in prediction, such that each input rows
+        has a row for each label and its probability score
+
+        :param model_df: Dataframe used in prediction
+        :param preds: List of prediction probabilities
+        :param labels: List of labels
+
+        :return: Prepared dataframe including input data and predictions
+        """
+        n_labels = len(labels)
+
+        # Repeat each row consecutively as the number of labels. At the end,
+        # the dataframe is expanded from (m, n) to (m*n_labels, n)
+        repeated_indexes = model_df.index.repeat(n_labels)
+        model_df = model_df.loc[repeated_indexes].reset_index(drop=True)
+
+        # Fill the dataframe with labels repeatedly, such that each input rows
+        # has a row for each label
+        extension_factor = model_df.shape[0]//n_labels
```

Comment:
This should be the size of the df before the extension, if I am not mistaken. Maybe, store it in a variable input_row_count before the extension happens